_rules/document-has-headings-047fe0.md

```diff
@@ -0,0 +1,306 @@
+---
+id: 047fe0
+name: Document has headings
+rule_type: atomic
+description: |
+  This rule checks that each section of content starts with a heading
+accessibility_requirements:
+  wcag-technique:H69: # Providing heading elements at the beginning of each section of content
+    forConformance: false
+    failed: not satisfied
+    passed: satisfied
+    inapplicable: satisfied
+input_aspects:
+  - DOM Tree
+  - CSS styling
+acknowledgements:
+  authors:
+    - Jean-Yves Moyen
+---
+
+## Applicability
+
+This rule applies to any [HTML web page][].
+
+## Expectations
+
+For each [section of content][] in the test target, the first node in the [flat tree][] which is inside this [section of content][] and has a non-empty [accessible name][]:
+
+- has a [semantic role][] of `heading`; and
+- is [visible][]; and
+- is [included in the accessibility tree][].
+
+**Note**: Neither this rule, nor technique [H69: Providing heading elements at the beginning of each section of content][tech h69], expects the heading to accurately describe its corresponding section. However, having non descriptive headings fails [Success Criterion 2.4.6: Headings and Labels](https://www.w3.org/TR/WCAG21/#headings-and-labels)
+
+**Note**: Neither this rule, nor technique [H69: Providing heading elements at the beginning of each section of content][tech h69], expects that headings are correctly nested without skipping level. It is nonetheless recommended to nest headings hierarchically without skipping levels.
+
+## Assumptions
+
+This rule assumes that headings used to pass technique [H69: Providing heading elements at the beginning of each section of content][tech h69] have to be [visible][] and [included in the accessibility tree][] in order to be beneficial to all users whether they are using assistive technologies or not.
+
+## Accessibility Support
+
+_There are no major accessibility support issues known for this rule._
+
+**Note to reviewers**: In its current state, the definition of [section of content][] would include, e.g. an advertising sidebar as a specific section of content. However, it is not clear that the lack of heading on it would be a breach of H69 or SC 2.4.1… Idea?
```

Comment:
If headers where the only way to skip repeating content I would definitely fail a certain section of content not having a heading of their own. This would result in a user not trusting the headings to describe everything in their "part" and could cause them to not use the headings to skip content in fear of missing something.

Reply:
Would you fail G124 (and 2.4.1) if the list of links at the top of the page does not include a link to the advertising section?

Reply:
I would fail G124 if single certain section of content (like advertising) didn't have a link.

However, in most of these cases I would still pass SC 2.4.1 because one of those links is bound to be a link to the start of the main content which would pragmatically turn it into a skiplink.

Reply:
Interesting…
```html
<a href="#nav">Navigation</a><a href="#main">Main</a>
<nav id="nav">link link</nav>
<aside id="ads">Buy!</aside>
<main id="main">Lorem Ipsum</a>
```
* This does not pass G1 because the first focusable is not a link to the main content.
* This does not pass G123 because the link before the `nav` section goes directly to the `main` section, not just to the end of the `nav` section.
* This does not pass G124 because the set of links at the start does not contain a link to the `ads` section.

But your saying that you would let this pass SC 2.4.1?

---
What about
```html
<a href="#nav">Navigation</a>
<a href="#ads1">Ads 1</a> … <a href="#ads19">Ads 19</a>
<a href="#main">Main</a>
<nav id="nav">link link</nav>
<aside id="ads1">Buy!</aside> … <aside id="ads20">Buy!</aside>
<main id="main">Lorem Ipsum</a>
```
This is missing a link to `ads20`, so G124 is KO, and has so many tab-stop before the link to `main` that is doesn't sound so good to let it pass 2.4.1 anyway…

---
(trying to understand the reasoning and going into corner cases in order to fine-tune the rule…)

Reply:
//But your saying that you would let this pass SC 2.4.1?//

Yes. You can pass a SC without passing any of the sufficient techniques.